### PR TITLE
[Fleet] added `download_rate` mapping to `.fleet-agents` index

### PR DESCRIPTION
**Pull request by juliaElastic**
Related to https://github.com/elastic/kibana/issues/171943

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)? yes
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md)? yes
- If submitting code, have you built your formula locally prior to submission with `gradle check`? yes
- If submitting code, is your pull request against main? Unless there is a good reason otherwise, we prefer pull requests against main and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)? yes
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

File: x-pack/plugin/core/template-resources/src/main/resources/fleet-agents.json [hunk 1]
```diff
@@ -270,6 +270,9 @@
                 "download_percent": {
                   "type": "double"
                 },
+                "download_rate": {
+                  "type": "double"
+                },
```
Comment (cmacknz) [inline x-pack/plugin/core/template-resources/src/main/resources/fleet-agents.json:275]: We also added two more download fields `retry_error_msg` and `retry_until`, let's map those while we are in here:

They came from https://github.com/elastic/elastic-agent/pull/3845

Comment (cmacknz) [reply x-pack/plugin/core/template-resources/src/main/resources/fleet-agents.json:275]: Relates https://github.com/elastic/kibana/issues/173370

Comment (juliaElastic) [reply x-pack/plugin/core/template-resources/src/main/resources/fleet-agents.json:275]: added here: https://github.com/elastic/elasticsearch/pull/103566/commits/9f2307516338d2199e2947b39a89563d28de7a0b

Comment (cmacknz) [reply x-pack/plugin/core/template-resources/src/main/resources/fleet-agents.json:275]: Thanks!

**Review by cmacknz — APPROVED** [review]

**Review by nchaulet — APPROVED** [review]

**Pull request merged**